File: CloudFlow/FrontEnd-App/src/app/shared/model/reports/actiuni-pe-proiect-task-report.model.ts
import { JsonObject, JsonProperty } from "json2typescript";
import { JsonDateConverter } from "../../json-mapper";
import { ObjectUtils } from "@app/shared/utils";
import { ReportConstants } from "@app/client/reports/constants/report.constants";

@JsonObject
export class ActiuniPeProiectTaskReportModel {

	@JsonProperty("numeProiect", String)
	public numeProiect: string = null;

	@JsonProperty("tipActiune", String)
	public tipActiune: string = null;
	
	public get tipActiuneForGrouping(): string {
		if (ObjectUtils.isNullOrUndefined(this.tipActiune)) {
			return ReportConstants.ACTIUN_PE_PROIECT_TIP_ACTIUNE_NULL_VALUE_FOR_GROUPING;
		}
		return this.tipActiune;
	}

	@JsonProperty("actiuni", String)
	public actiuni: string = null;

	@JsonProperty("dataInceputTask", JsonDateConverter)
	public dataInceputTask: Date = null;

	@JsonProperty("status", String)
	public status: string = null;

	@JsonProperty("numeSubproiect", String)
	public numeSubproiect: string = null;
}

File: CloudFlow/src/ro/cloudSoft/cloudDoc/presentation/client/shared/model/report/ActiuniPeProiectTaskReportModel.java
package ro.cloudSoft.cloudDoc.presentation.client.shared.model.report;

import java.util.Date;

public class ActiuniPeProiectTaskReportModel {

	private String numeProiect;
	private String numeSubproiect;
	private String tipActiune;
	private String actiuni;
	private Date dataInceputTask;
	private String status;
	
	public String getNumeProiect() {
		return numeProiect;
	}
	
	public void setNumeProiect(String numeProiect) {
		this.numeProiect = numeProiect;
	}
	
	public String getTipActiune() {
		return tipActiune;
	}
	
	public void setTipActiune(String tipActiune) {
		this.tipActiune = tipActiune;
	}
	
	public String getActiuni() {
		return actiuni;
	}
	
	public void setActiuni(String actiuni) {
		this.actiuni = actiuni;
	}
	
	public Date getDataInceputTask() {
		return dataInceputTask;
	}
	
	public void setDataInceputTask(Date dataInceputTask) {
		this.dataInceputTask = dataInceputTask;
	}
	
	public String getStatus() {
		return status;
	}
	
	public void setStatus(String status) {
		this.status = status;
	}

	public String getNumeSubproiect() {
		return numeSubproiect;
	}

	public void setNumeSubproiect(String numeSubproiect) {
		this.numeSubproiect = numeSubproiect;
	}
	
}
